assets/js/serialized-input.js

```diff
@@ -11,16 +11,34 @@ export default function ( parent, hiddenInput ) {
 
 	};
 
+	const checkLimit = ( currentValue ) => {
+		var count = currentValue.length;
+		var field = parent.querySelector( '.customizer-curation-list-input' );
+		var max = field.getAttribute( 'data-max' );
+
+		if ( max ) {
```

Comment:
I feel like this should be handled in the autosuggest-list level. Essentially the serialized input doesn't really care about the add button. Its job is only to control adding and remove items from the serialized input.